templates/workspace_services/mlflow/terraform/postgresql.tf

```diff
@@ -0,0 +1,57 @@
+resource "random_string" "username" {
+  length      = 10
+  upper       = true
+  lower       = true
+  number      = true
+  min_numeric = 1
+  min_lower   = 1
+  special     = false
+}
+
+resource "random_password" "password" {
+  length           = 16
+  lower            = true
+  min_lower        = 1
+  upper            = true
+  min_upper        = 1
+  number           = true
+  min_numeric      = 1
+  special          = true
+  min_special      = 1
+  override_special = "_%@"
+}
+
+resource "azurerm_key_vault_secret" "postgresql_admin_password" {
+  name         = "${local.webapp_name}-admin-credentials"
+  value        = "${random_string.username.result}\n${random_password.password.result}"
+  key_vault_id = data.azurerm_key_vault.ws.id
+}
+
+resource "azurerm_postgresql_server" "mlflow" {
+  name                = local.postgresql_server_name
+  location            = data.azurerm_resource_group.ws.location
+  resource_group_name = data.azurerm_resource_group.ws.name
+
+  administrator_login          = random_string.username.result
+  administrator_login_password = random_password.password.result
+
+  sku_name   = "GP_Gen5_4"
```

Comment:
That is $300/month...
IIRC, MLFlow is pretty lite on DB requirements as most things are stored in the file system. Are you sure about this size?

Reply:
Fair point. I was not sure about the size and storage requirements and hence I just put GP with 4 core and 512 GB. I have changed it to Basic with 1 core and 5120 MB. With auto grow set to true storage shouldn't be a problem.